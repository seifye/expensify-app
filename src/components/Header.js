// import React for using JSX
import React from 'react'

/* NavLink can be used to do client side routing behind the scenes without
a full page refresh to navigate the different pages in our app */
import { NavLink } from 'react-router-dom'



const Header = () => (
    <header>
        <h1> Expensify App </h1>
        <NavLink to='/' activeClassName='is-active' exact={true}> Dashboard </NavLink>
        <NavLink to='/create' activeClassName='is-active'> Create Expense </NavLink>
        <NavLink to='/help' activeClassName='is-active'> Help </NavLink>
    </header>
)

export default Header