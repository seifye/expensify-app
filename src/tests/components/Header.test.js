// import React for using JSX
import React from 'react'

// import other js files => use
import Header from '../../components/Header'

/* There are two main ways to test react components (Shallow rendering and full DOM rendering). 
Shallow rendering renders only a given component while full DOM rendering renders child components as
well */

// Importing the shallow renderer from enzyme library
import { shallow } from 'enzyme'





// Testing React Component
// This is a static component. It doesn't have any state, it doesn't take in any props
test('Should render Header component correctly', () => {
    const wrapper = shallow(<Header />)
    /* Snapshots allow us to track changes to data over time. We will get notified if the rendered
    component ever changes. So if the component output changed in a way we don't want, we can catch and
    fix that. In contrast if it changed in a way we do want, we can allow that by updating the snapshot. 
    The last snapshot is always stored in the _snapshots_ folder which is used in comparison with the
    new snapshot. If you accepted the update (by pressing 'u') then the old snapshot in the  _snapshot_
    folder gets replaced by the new snapshot */
    expect(wrapper).toMatchSnapshot()
})




