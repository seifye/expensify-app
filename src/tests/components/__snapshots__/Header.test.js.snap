// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header component correctly 1`] = `
<header>
  <h1>
     Expensify App 
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
     Dashboard 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
     Create Expense 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
     Help 
  </NavLink>
</header>
`;
